cpp/src/io/avro/avro_gpu.cu

```diff
@@ -250,22 +257,13 @@ extern "C" __global__ void __launch_bounds__(NWARPS * 32, 2)
 
   // Fetch schema into shared mem if possible
   if (schema_len <= MAX_SHARED_SCHEMA_LEN) {
-    for (int i = threadIdx.y * 32 + threadIdx.x;
-         i < schema_len * sizeof(schemadesc_s) / sizeof(uint32_t);
-         i += NWARPS * 32) {
-      reinterpret_cast<uint32_t *>(&g_shared_schema)[i] =
-        reinterpret_cast<const uint32_t *>(schema_g)[i];
-    }
-    __syncthreads();
+    cuda::memcpy_async(g_shared_schema, schema_g, schema_len * sizeof(schemadesc_s), barrier);
```

Comment:
This doesn't look right. The original code copied from `schema_g` cooperatively. This code has every thread redundantly copying the entire `schema_g` array. In order to cooperatively copy you need to pass the `group` as the first argument.

Reply:
Ahh, but passing group is not yet supported as it is depending on #6275 .

Reply:
How about using `memcpy` for now with thread 0 and sync. Once #6275 is merged, we can update this.

Reply:
@jrhemstad Else we can leave it as it before this change and update it once #6275 is merged.

Reply:
You don't want to use normal `memcpy` here. You'd be better off just doing normal element-wise assignment in parallel.